content/en/blog/2022/k8s-otel-expose.md

```diff
@@ -0,0 +1,308 @@
+spelling:
+  cSpell:ignore k8sattributes k8sattributesprocessor K8sattributes k8sprocessor
+  cSpell:ignore K8sprocessor KUBE
+---
+
+Exposing a [OpenTelemetry Collector](https://opentelemetry.io/docs/collector/) currently requires a number of configuration steps. Goal of this blog post is to demonstrate how to establish a secure communication between two collectors in different kubernetes clusters.
+
+
+Details of CRDs and dependency installations are not covered by this post.
+
+# Overview
+
+The following graph illustrates the target topology. Goal is to transfer traces generated by a [test application](https://github.com/frzifus/jaeger-otel-test/pkgs/container/jaeger-otel-test) via a dedicated collector to a publicly accessible cluster. The receiving collector uses the transmitted ['Basic' HTTP Authentication](https://datatracker.ietf.org/doc/html/rfc7617) credentials to check whether the sender is authorized to store data. Finally, transmitted traces are stored in a [Jaeger in-memory](https://www.jaegertracing.io/docs/1.37/deployment/#memory) database for easy verification.
+
+```mermaid
+sequenceDiagram
+    participant Application
+    participant Edge Collector
+    participant Remote Collector
+    participant Jaeger Backend
+
+
+    Note over Application,Jaeger Backend: <<== Edge Cluster | | Public Cluster ==>>
+
+    loop
+        Application->>Application: generate
+    end
+
+    Application->>Edge Collector: (UDP) Compact thrift protocol
+    activate Edge Collector
+    Edge Collector->>Edge Collector: Insert metadata
+    Edge Collector->>Edge Collector: Insert basicauth credentials
+    Edge Collector->>Remote Collector: (gRPC) Traces serialized with protobuf
+    deactivate Edge Collector
+    activate Remote Collector
+    Remote Collector->>Remote Collector: Verify basicauth credentials
+    Remote Collector->>Jaeger Backend: (UDP) Compact thrift protocol
+    deactivate Remote Collector
+
+```
+
+# Prerequisites
+
+Interfaces and behavior may change in the future. Therefore, the version used in this setup is mentioned in brackets.
+
+- A Kubernetes[v1.23.3] cluster with a public address with installed [ingress-nginx-controller](https://docs.nginx.com/nginx-ingress-controller/)[v1.2.1].
+- A Kubernetes[v1.23.3] edge cluster, to create a test cluster, using [Kind](https://kind.sigs.k8s.io/) is recommended.
+- Installed [OpenTelemetry Operator](https://opentelemetry.io/docs/collector/getting-started/#kubernetes)[v0.58.0] on both ends.
+- Installed [Jaeger Operator](https://www.jaegertracing.io/docs/1.37/operator/)[v1.37.0] on your public cluster.
+- Installed [cert-manager](https://cert-manager.io/)[v1.9.1] on your public cluster.
+
+# Remote cluster configuration
+
+Since all components except the Jaeger backend dependent on a following component, we begin by deploying the backend.
+
+```yaml
+apiVersion: jaegertracing.io/v1
+kind: Jaeger
+metadata:
+  name: my-in-memory
+```
+
```

Comment:
It would be nice, if possible, if you could add the little widget that allows clicking on a button to copy code in these examples.

Reply:
How does that work?

Reply:
Hmmm...I don't know, actually.  :-)  I've seen it a few other places.  I'll try to look into it.

Reply:
this would require a change in the hugo template, we have talked about this in the past, but it's not there yet.